server/sonar-server/src/main/java/org/sonar/server/platform/ServerId.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.platform;
+
+public class ServerId {
```

Comment:
I didn't find any clear answer. But from what I read, this annotation concerns the design of the annotated class, not the object, so, it's not a problem the class is not final.

https://books.google.fr/books?id=EK43StEVfJIC&pg=PA353&lpg=PA353&dq=java+should+class+annotated+with+@Immutable+be+final&source=bl&ots=uoZAB7qQou&sig=TkjMUg-ntUQr0lLM8IgflISWbnE&hl=fr&sa=X&ved=0ahUKEwjijKK1nOPMAhWKvRoKHaRIC7YQ6AEIVzAH#v=onepage&q=java%20should%20class%20annotated%20with%20%40Immutable%20be%20final&f=false

Imho, this defensive approach of making the class final is not _required_ in this specific case (it would be required in an API) but in the other hand, making the class final won't hurt, so, I'll make it final.